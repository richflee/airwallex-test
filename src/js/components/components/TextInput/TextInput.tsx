import * as React from 'react';
import styled from 'styled-components';

const StyledTextInput = styled.input`
    border: 1px solid #6ba8a9;
    border-radius: 0.25em;
`;

export default class TextInput extends React.Component<any, any> {
    constructor(props: any) {
        super(props);
    }

    render() {
        return (
            <StyledTextInput
                placeholder={this.props.placeholder}
                type={this.props.type}
                name={this.props.name}
                id={this.props.id}
                value={this.props.value}
                onChange={this.props.handleInputChange} />
        );
    }
}
