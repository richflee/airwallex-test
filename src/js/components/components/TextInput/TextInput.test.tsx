import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TextInput from './TextInput';

describe("TextInput component", () => {
    test("Matches the snapshot", () => {
        const value = 'foo';
        const textInput = renderer.create(<TextInput  placeholder="Full name" type="text" name="Full name" id="fullName" value={value} handleInputChange={() => {return true;}} />);
        expect(textInput.toJSON()).toMatchSnapshot();
    });
});