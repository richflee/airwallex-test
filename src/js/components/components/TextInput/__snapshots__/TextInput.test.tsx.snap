// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component Matches the snapshot 1`] = `
<input
  className="sc-bdVaJa joulRW"
  id="fullName"
  name="Full name"
  onChange={[Function]}
  placeholder="Full name"
  type="text"
  value="foo"
/>
`;
